proto-buf/services/eigentrust.proto

```diff
@@ -0,0 +1,285 @@
+  // 11438 | [11000..12000) | yes       | 11000
+  // 11975 | [11000..12000) | no        |
+  // 11999 | [11000..12000) | no        |
+  // 12000 | [12000..13000) | yes       | 12000
+  // 12014 | [12000..13000) | no        |
+  //
+  // (Note that the result for timestamp=12000
+  // does NOT reflect the triggering input at timestamp=12000.)
+  repeated uint64 period_qwords = 2;
+}
+
+message CreateTrustMatrixRequest {
+}
+
+message CreateTrustMatrixResponse {
+  string id = 1;
+}
+
+message GetTrustMatrixRequest {
+  string id = 1;
+}
+
+message GetTrustMatrixResponse {
+  oneof part {
+    TrustMatrixHeader header = 1;
+    TrustMatrixEntry entry = 2;
+  }
+}
+
+message UpdateTrustMatrixRequest {
+  TrustMatrixHeader header = 1;
+  repeated TrustMatrixEntry entries = 2;
+}
+
+message UpdateTrustMatrixResponse {
+}
+
+message FlushTrustMatrixRequest {
+  string id = 1;
+}
+
+message FlushTrustMatrixResponse {
+}
+
+message DeleteTrustMatrixRequest {
+  string id = 1;
+}
+
+message DeleteTrustMatrixResponse {
+}
+
+message CreateTrustVectorRequest {
+}
+
+message CreateTrustVectorResponse {
+  string id = 1;
+}
+
+message GetTrustVectorRequest {
+  string id = 1;
+}
+
+message GetTrustVectorResponse {
+  oneof part {
+    TrustVectorHeader header = 1;
+    TrustVectorEntry entry = 2;
+  }
+}
+
+message UpdateTrustVectorRequest {
+  TrustVectorHeader header = 1;
+  repeated TrustVectorEntry entries = 2;
+}
+
+message UpdateTrustVectorResponse {
+}
+
+message FlushTrustVectorRequest {
+  string id = 1;
+}
+
+message FlushTrustVectorResponse {
+}
+
+message DeleteTrustVectorRequest {
+  string id = 1;
+}
+
+message DeleteTrustVectorResponse {
+}
+
+message BasicComputeRequest {
+  ComputeParams params = 1;
+}
+
+message BasicComputeResponse {
+  // TODO(ek): Add flat-tail
+}
+
+message CreateComputeJobRequest {
+  ComputeJobSpec spec = 1;
+}
+
+message CreateComputeJobResponse {
+  string id = 1;
+}
+
+message DeleteComputeJobRequest {
+  string id = 1;
+}
+
+message DeleteComputeJobResponse {
+}
+
+service Compute {
+  // Create a new trust matrix (for local trust), return its ID.
+  rpc CreateTrustMatrix(CreateTrustMatrixRequest)
+      returns (CreateTrustMatrixResponse) {}
+
+  // Get the full contents of a trust matrix.
+  // The header is sent first (includes metadata such as timestamp),
+  // followed by non-zero entries.
+  rpc GetTrustMatrix(GetTrustMatrixRequest)
+      returns (stream GetTrustMatrixResponse) {}
+
+  // Update a trust matrix.
+  // Entries for the same matrix with the same timestamp can be batched
+  // in the same request.
+  // Updates cannot go back in time:
+  // An update older than the target matrix is rejected.
+  // Updates can be empty (have zero entries):
+  // This can be used to force-update the timestamp of the trust matrix
+  // and trigger periodic re-compute of a compute job.
+  rpc UpdateTrustMatrix(UpdateTrustMatrixRequest)
+      returns (UpdateTrustMatrixResponse) {}
```

Comment:
So I will have to use this to push LT matrix items into the core compute?
Is there a logic inside CoreCompute to pull the terms from LC component?

Reply:
Yes, that pulling was what I've doing so far.  Pushing LT entries as they are updated streamlines processing more evenly though.

Reply:
So I should think about how to use this to push terms to CoreCompute for this milestone as well?

BTW, what does P1 mean?

Reply:
No, not for this milestone.  I can continue using the puller.  That's what I meant by P1 (priority 1), as opposed to P0 (priority 0, i.e. must be done this milestone). :-)

Reply:
I don't want to have the pulling logic in the long-running system such as Milestone 2. :-p